### DISCUSSION TITLE
Enabling capacity metric collection

### DISCUSSION THREAD

**Discussion by edd1619** [NetApp/harvest#1556] [Q&A] [answered]
Hello,

Could you direct me to the documentation on enabling capacity metric collection?

Thanks!

**Comment by cgrinds** [comment]
hi @edd1619 do you mean the metrics shown under the `Capacity Metrics` row shown below?
If so, those are enabled by default as part of the `Volume` and `Aggregate` templates in the Zapi or Rest collector.
https://github.com/NetApp/harvest/blob/main/conf/zapi/default.yaml#L9

![image](https://user-images.githubusercontent.com/242252/204346075-4e198e31-76ee-4f38-a870-0129971d99af.png)

**Comment by edd1619** [comment]
Hi @cgrinds

We are looking for per aggregate/volume/qtree/lun capacity. When we check Prometheus, we only see disk capacity for aggr and only perf metrics from volumes. Do we have to enable per aggregate/volume/qtree/lun capacity metric somehow ??
***@***.***

Thanks!
@edd1619<https://github.com/edd1619>


From: Chris Grindstaff ***@***.***>
Sent: Monday, November 28, 2022 12:52 PM
To: NetApp/harvest ***@***.***>
Cc: Dannenfelser, Edward ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [NetApp/harvest] Enabling capacity metric collection (Discussion #1556)


hi @edd1619<https://github.com/edd1619> do you mean the metrics shown under the Capacity Metrics row shown below?
If so, those are enabled by default as part of the Volume and Aggregate templates in the Zapi or Rest collector.
https://github.com/NetApp/harvest/blob/main/conf/zapi/default.yaml#L9<https://github.com/NetApp/harvest/blob/main/conf/zapi/default.yaml#L9>

[image]<https://user-images.githubusercontent.com/242252/204346075-4e198e31-76ee-4f38-a870-0129971d99af.png>

—
Reply to this email directly, view it on GitHub<https://github.com/NetApp/harvest/discussions/1556#discussioncomment-4255889>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOGQCXVMRK54XFTBSBE3UGLWKTWLTANCNFSM6AAAAAASNRLVZI>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>
<br /><br />
ICON plc made the following annotations.
------------------------------------------------------------------------------
This e-mail transmission may contain confidential or legally privileged information that is intended only for the individual or entity named in the e-mail address. If you
are not the intended recipient, you are hereby notified that any disclosure, copying, distribution, or reliance upon the contents of this e-mail is strictly prohibited. If
you have received this e-mail transmission in error, please reply to the sender, so that ICON plc can arrange for proper delivery, and then please delete the message.

Thank You,

ICON plc
South County Business Park
Leopardstown
Dublin 18
Ireland
Registered number: 145835

**Comment by cgrinds** [comment]
hi @edd1619 what version of Harvest are you using? 
I see capacity information in the Harvest `22.11` dashboards - so trying to understand if we're talking about different things?

1. Do you see the data you need in one of Harvest's dashboards?
2. Or, you aren't using the out-of-the-box dashboards, and you are instead querying Prometheus directly or via your own dashboards? 
3. Can you share one or two prom queries you expect to return data that aren't? 
4. Did these metrics exist earlier, and now they don't, or this is the first time you're trying to collect them?

**Comment by edd1619** [comment, marked as answer]
Hi Chris,

We are on version 21.08.

We see LUN size/size used data in the dashboard but no Volume, Aggregate or Qtree size/size used data.

WeE are unable to create the query because we do not see ‘aggr_size’ or ‘aggr_size_used’ metric when trying to create the query, we only see ‘aggr_disk….’ options. We should have the 2 previous mentioned metrics as well??
***@***.***

These did not exist. We are just moving to Harvest 2.0 so it is new setup.

Thanks!

From: Chris Grindstaff ***@***.***>
Sent: Monday, November 28, 2022 2:42 PM
To: NetApp/harvest ***@***.***>
Cc: Dannenfelser, Edward ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [NetApp/harvest] Enabling capacity metric collection (Discussion #1556)


hi @edd1619<https://github.com/edd1619> what version of Harvest are you using?
I see capacity information in the Harvest 22.11 dashboards - so trying to understand if we're talking about different things?

  1.  Do you see the data you need in one of Harvest's dashboards?
  2.  Or, you aren't using the out-of-the-box dashboards, and you are instead querying Prometheus directly or via your own dashboards?
  3.  Can you share one or two prom queries you expect to return data that aren't?
  4.  Did these metrics exist earlier, and now they don't, or this is the first time you're trying to collect them?

—
Reply to this email directly, view it on GitHub<https://github.com/NetApp/harvest/discussions/1556#discussioncomment-4256691>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOGQCXSQOIQVO6U5BW7TJE3WKUDH5ANCNFSM6AAAAAASNRLVZI>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>
<br /><br />
ICON plc made the following annotations.
------------------------------------------------------------------------------
This e-mail transmission may contain confidential or legally privileged information that is intended only for the individual or entity named in the e-mail address. If you
are not the intended recipient, you are hereby notified that any disclosure, copying, distribution, or reliance upon the contents of this e-mail is strictly prohibited. If
you have received this e-mail transmission in error, please reply to the sender, so that ICON plc can arrange for proper delivery, and then please delete the message.

Thank You,

ICON plc
South County Business Park
Leopardstown
Dublin 18
Ireland
Registered number: 145835

**Reply by cgrinds** [reply]
Thanks! That explains why we're seeing different things. That version is quite old. We've had six releases since then with many bug fixes and new features. Can you upgrade to [22.11](https://github.com/NetApp/harvest/releases/tag/v22.11.0) and take a look at the new dashboards that should include what you're looking for.

There are several dashboards that use the metrics under `aggr_space_*`, `volume_size`, `volume_space_physical_used`, `volume_space_logical_used`, `volume_size_used`.

It's probably best to upgrade, import the Harvest dashboards, and take a look at the capacity panels already exposed in `Aggregate`, `Disk`, `Lun`, `Qtree`, and `Volume`. Those panels can be inspected for their prom query and metrics.

**Reply by Dinesvl** [reply]
Hi Cris,

Could you please give me link for 22.11 version .rpm offline installation files for redhat? I tried installing 22.11 and failed to download some repo's while installing. I don't have internet in the client machine which i am trying to upgrade. Thanks

**Comment by edd1619** [comment]
Great thank you! We will try upgrade.

On the upgrade topic, we tried an upgrade in test and it was not too smooth. Is there a detailed upgrade link you could provide?

Thanks!

From: Chris Grindstaff ***@***.***>
Sent: Monday, November 28, 2022 3:20 PM
To: NetApp/harvest ***@***.***>
Cc: Dannenfelser, Edward ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [NetApp/harvest] Enabling capacity metric collection (Discussion #1556)


Thanks! That explains why we're seeing different things. That version is quite old. We've had six releases since then with many bug fixes and new features. Can you upgrade to 22.11<https://github.com/NetApp/harvest/releases/tag/v22.11.0> and take a look at the new dashboards that should include what you're looking for.

There are several dashboards that use the metrics under aggr_space_*, volume_size, volume_space_physical_used, volume_space_logical_used, volume_size_used.

It's probably best to upgrade, import the Harvest dashboards, and take a look at the capacity panels already exposed in Aggregate, Disk, Lun, Qtree, and Volume. Those panels can be inspected for their prom query and metrics.

—
Reply to this email directly, view it on GitHub<https://github.com/NetApp/harvest/discussions/1556#discussioncomment-4256972>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOGQCXUVQOOGLT3ZJTXN2YTWKUHV7ANCNFSM6AAAAAASNRLVZI>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>
<br /><br />
ICON plc made the following annotations.
------------------------------------------------------------------------------
This e-mail transmission may contain confidential or legally privileged information that is intended only for the individual or entity named in the e-mail address. If you
are not the intended recipient, you are hereby notified that any disclosure, copying, distribution, or reliance upon the contents of this e-mail is strictly prohibited. If
you have received this e-mail transmission in error, please reply to the sender, so that ICON plc can arrange for proper delivery, and then please delete the message.

Thank You,

ICON plc
South County Business Park
Leopardstown
Dublin 18
Ireland
Registered number: 145835

**Reply by cgrinds** [reply]
hi @edd1619 take a look at https://netapp.github.io/harvest/22.11/upgrade/ and if you have problems, hit us up on [Discord](https://discordapp.com/channels/855068651522490400/1001963189124206732).

**Reply by Dinesvl** [reply]
Hi Cris,

Could you please give me link for 22.11 version .rpm offline installation files for redhat? I tried installing 22.11 and failed to download some repo's while installing. I don't have internet in the client machine which i am trying to upgrade. Thanks

**Reply by rahulguptajss** [reply]
@Dinesvl You can download latest release artifacts from here https://github.com/NetApp/harvest/releases/latest

Direct rpm link for 22.11 https://github.com/NetApp/harvest/releases/download/v22.11.0/harvest-22.11.0-1.x86_64.rpm

**Reply by Dinesvl** [reply]
Thanks for the download link. i exported proxy setting in redhat machine to download repo and then completed upgrade.

**Reply by Dinesvl** [reply]
Hi Cris \ Rahul,

After upgraded to 2.0, I found the false spikes in "latency from cluster" and thorughput metrics. some of the spikes are different from version 1.6 and 2.0. Did you face any issues like this ?

**Reply by cgrinds** [reply]
hi @Dinesvl yes we have seen that and it's an area we've worked to improve between 1.6 and 2.0. The short version of what is going on is sometimes ONTAP does not return performance metrics and because the performance metrics are raw values that Harvest must delta with the previous poll, this can cause spikes since to Harvest, you had a large raw value during the previous poll and now you have a zero, which causes large deltas. Harvest 2 does a better job of detecting these cases and not publishing bad data.

There were some cases where Harvest 1.6 would suppress some zero values and Harvest 2.0 does not. 

Can you share a screenshot of Harvest 2 vs 1.6 and the false spike you're seeing?

**Reply by Dinesvl** [reply]
![harvest2 0](https://user-images.githubusercontent.com/119581283/206962786-35e62ada-c58b-429b-8fcd-b97d9123bbfa.PNG)
![harvest 1 6_screenshot](https://user-images.githubusercontent.com/119581283/206962806-5d684b33-9413-47fd-bbc1-a315db67072a.PNG)


Hi Cris,

Attached is the screenshot of both 2.0 and 1.6 versions latency from cluster output. We see one of the volume have 13.3 min latency spike from 2.0 which is not in 1.6 for same period of time.

**Reply by rahulguptajss** [reply]
@Dinesvl Do these spikes happening frequently? Do you see such spikes for other metrics as well? We recently have handled some of the corner cases which are currently available through our [nightly](https://github.com/NetApp/harvest/releases/tag/nightly) release. Could you give that a try and let us know the outcome.

**Reply by Dinesvl** [reply]
@rahulguptajss yes, its happening frequently. Other metrics like svm latency and throughputs also having same issues.

Let me upgrade as you said and get back on this.

**Reply by Dinesvl** [reply]
@rahulguptajss .. it says can not load RPM files and time out

Updating Subscription Management repositories.
Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)                                                                                 48 kB/s | 2.8 kB     00:00
Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)                                                                                    48 kB/s | 2.4 kB     00:00
Can not load RPM file: harvest-22.12.12-nightly.x86_64.rpm.

**Reply by rahulguptajss** [reply]
@Dinesvl I suspect rpm downloaded on your end got corrupt. Could you share the command, you are running. I tried below and works fine.

`rpm -i https://github.com/NetApp/harvest/releases/download/nightly/harvest-22.12.13-nightly.x86_64.rpm`

**Reply by Dinesvl** [reply]
@rahulguptajss 
I used below commands.
yum install harvest-22.12.12-nightly.x86_64.rpm
yum localinstall harvest-22.12.12-nightly.x86_64.rpm

Also no response when i tried rpm -i command which you gave

**Reply by rahulguptajss** [reply]
Let's try to see if rpm downloaded is not corrupted. Could you share output of below commands

`rpm --checksig harvest-22.12.13-nightly.x86_64.rpm`
`rpm -qip harvest-22.12.13-nightly.x86_64.rpm`
`rpm -qa | grep harvest`

Note that today nightly rpm name is `harvest-22.12.13-nightly.x86_64.rpm`

**Reply by Dinesvl** [reply]
I tried with RPM name harvest-22.12.13-nightly.x86_64.rpm.. and it worked fine. Can i monitor the spikes for 1 more day ?

**Reply by rahulguptajss** [reply]
cool. Yes let's monitor it.

**Comment by edd1619** [comment]
Great thanks for the help Chris!


From: Chris Grindstaff ***@***.***>
Sent: Monday, November 28, 2022 3:33 PM
To: NetApp/harvest ***@***.***>
Cc: Dannenfelser, Edward ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [NetApp/harvest] Enabling capacity metric collection (Discussion #1556)


hi @edd1619<https://github.com/edd1619> take a look at https://netapp.github.io/harvest/22.11/upgrade/<https://netapp.github.io/harvest/22.11/upgrade/> and if you have problems, hit us up on Discord<https://discordapp.com/channels/855068651522490400/1001963189124206732>.

—
Reply to this email directly, view it on GitHub<https://github.com/NetApp/harvest/discussions/1556#discussioncomment-4257089>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOGQCXW4GUMQXDH4AAKZF23WKUJH7ANCNFSM6AAAAAASNRLVZI>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>
<br /><br />
ICON plc made the following annotations.
------------------------------------------------------------------------------
This e-mail transmission may contain confidential or legally privileged information that is intended only for the individual or entity named in the e-mail address. If you
are not the intended recipient, you are hereby notified that any disclosure, copying, distribution, or reliance upon the contents of this e-mail is strictly prohibited. If
you have received this e-mail transmission in error, please reply to the sender, so that ICON plc can arrange for proper delivery, and then please delete the message.

Thank You,

ICON plc
South County Business Park
Leopardstown
Dublin 18
Ireland
Registered number: 145835
